### PR TITLE
mydumper: support estimate count for TiDB

### PR DESCRIPTION
**Pull request by csuzhangxc**
support use `EXPLAIN SELECT` get estimate count from TiDB

ref: https://github.com/pingcap/tidb/blob/09beefbe045011e3c77608c9ed33da87c11efa94/planner/core/common_plans.go#L486

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@GregoryIan @kennytm PTAL

File: mydumper.c [hunk 1]
```diff
@@ -2043,8 +2043,23 @@ guint64 estimate_count(MYSQL *conn, char *database, char *table, char *field, ch
 	MYSQL_FIELD * fields = mysql_fetch_fields(result);
 
 	guint i;
+	const char* field_name = "rows";
+	if (detected_server == SERVER_TYPE_TIDB) {
+		// result of `EXPLAIN SELECT` for TiDB is different than MySQL, we fetch the first row's `count` is enough.
+		// the string format of `count` is `f`, so we can treat it as guint64.
```
Comment (gregwebs) [inline mydumper.c]: guint64 -> uint64

Comment (csuzhangxc) [reply mydumper.c]: :open_mouth: but we use `guint64 count=0;` in L2068

**Comment by GregoryIan** [comment]
LGTM

**Pull request merged**